libs/libcommon/src/libcommon/state.py

```diff
@@ -345,204 +245,67 @@ def __post_init__(self) -> None:
-                # ^ safety check
+            self.artifact_state_by_step = {
+                processing_step.name: ArtifactState(
+                    processing_step=processing_step,
+                    dataset=self.dataset,
+                    revision=self.revision,
+                    config=None,
+                    split=None,
+                    error_codes_to_retry=self.error_codes_to_retry,
+                    pending_jobs_df=self.pending_jobs_df[
+                        (self.pending_jobs_df["revision"] == self.revision)
+                        & (self.pending_jobs_df["config"].isnull())
+                        & (self.pending_jobs_df["split"].isnull())
+                        & (self.pending_jobs_df["type"] == processing_step.job_type)
+                    ],
+                    cache_entries_df=self.cache_entries_df[
+                        (self.cache_entries_df["kind"] == processing_step.cache_kind)
+                        & (self.cache_entries_df["config"].isnull())
+                        & (self.cache_entries_df["split"].isnull())
+                    ],
+                )
+                for processing_step in self.processing_graph.get_input_type_processing_steps(input_type="dataset")
+            }
 
             with StepProfiler(
-                method="DatasetState.__post_init__",
-                step="get_dataset_level_artifact_states",
-                context=f"dataset={self.dataset}",
-            ):
-                self.artifact_state_by_step = {
-                    processing_step.name: ArtifactState(
-                        processing_step=processing_step,
-                        dataset=self.dataset,
-                        revision=self.revision,
-                        config=None,
-                        split=None,
-                        error_codes_to_retry=self.error_codes_to_retry,
-                        pending_jobs_df=self.pending_jobs_df[
-                            (self.pending_jobs_df["revision"] == self.revision)
-                            & (self.pending_jobs_df["config"].isnull())
-                            & (self.pending_jobs_df["split"].isnull())
-                            & (self.pending_jobs_df["type"] == processing_step.job_type)
-                        ],
-                        cache_entries_df=self.cache_entries_df[
-                            (self.cache_entries_df["kind"] == processing_step.cache_kind)
-                            & (self.cache_entries_df["config"].isnull())
-                            & (self.cache_entries_df["split"].isnull())
-                        ],
-                    )
-                    for processing_step in self.processing_graph.get_input_type_processing_steps(input_type="dataset")
-                }
-            with StepProfiler(
-                method="DatasetState.__post_init__",
+                method="DatasetBackfillPlan.__post_init__",
```

Comment:
I don't understand, should we use in the method the name of the current class + method? Why it changed to DatasetBackfillPlan?

Reply:
It's an error!